source/extensions/transport_sockets/tls/context_impl.cc

```diff
@@ -1171,6 +1171,11 @@ bool ContextImpl::verifyCertChain(X509& leaf_cert, STACK_OF(X509) & intermediate
     error_details = "Failed to verify certificate chain: X509_STORE_CTX_init";
     return false;
   }
+  // Currently this method is only used to verify server certs, so hard-code "ssl_server" for now.
+  if (!X509_STORE_CTX_set_default(ctx.get(), "ssl_server")) {
+    error_details = "Failed to verify certificate chain: X509_STORE_CTX_set_default";
+    return false;
```

Comment:
TBH, it's not clear to me how X509_STORE_CTX_set_default() would return 0 with "ssl_server". Probably this is impossible if the name is hard-coded to "ssl_server". But I'm not a boring SSL expert.

Reply:
Even if it is impossible with the hard-coded parameter, I would still prefer to handle the case where it returns 0 so that future BoringSSL change doesn't break Envoy in a unexpected way.

Reply:
Get it. But it seems that without this test, the test coverage is not up to the requirements. We can let the maintainer decide  whether we can reduce the coverage requirement of this module.

Reply:
added a few irrelevant tls unit tests to raise coverage.